backend/src/main/java/gov/cdc/usds/simplereport/config/simplereport/DemoUserConfiguration.java

```diff
@@ -12,30 +15,41 @@
  * make demo login more useful.
  */
 @ConfigurationProperties(prefix = "simple-report.demo-users")
-@ConstructorBinding
 public class DemoUserConfiguration {
 
   private DemoUser defaultUser;
-  private List<DemoUser> alternateUsers;
+  private List<DemoUser> users;
+  private Map<String, DemoUser> byUsername;
 
+  public DemoUserConfiguration(List<DemoUser> allUsers) {
+    this(null, allUsers);
+  }
+
+  @ConstructorBinding
   public DemoUserConfiguration(DemoUser defaultUser, List<DemoUser> alternateUsers) {
     super();
     this.defaultUser = defaultUser;
-    this.alternateUsers = alternateUsers == null ? List.of() : alternateUsers;
+    this.users = new ArrayList<>();
+    if (defaultUser != null) {
+      users.add(defaultUser);
+    }
+    if (alternateUsers != null) {
+      users.addAll(alternateUsers);
+    }
+    byUsername =
+        users.stream().collect(Collectors.toMap(DemoUser::getUsername, Function.identity()));
   }
 
   public DemoUser getDefaultUser() {
     return defaultUser;
   }
 
-  public List<DemoUser> getAlternateUsers() {
-    return alternateUsers;
+  public List<DemoUser> getAllUsers() {
+    return users;
   }
 
-  public List<DemoUser> getAllUsers() {
-    List<DemoUser> allUsers = new ArrayList<>(getAlternateUsers());
-    allUsers.add(getDefaultUser());
-    return allUsers;
+  public DemoUser getByUsername(String username) {
```

Comment:
this smells off. i feel like the caller should be informed when they call `getByUsername()` that their chosen username does not exist in `DemoUserConfiguration` -- it shouldn't just silently return them the default user. i'd rather return null and then if the caller chooses to subsequently get the default user, they can do that.

Reply:
I agree, I think I was trying too hard to cut down on branches in the calling code.